### ISSUE TITLE
Significant lag from opening a Brush window (Win10)

### ISSUE THREAD

**Issue by Bobhostern** [mypaint/mypaint#516]
I get lag only when I have a brush window open. Whenever I start the program, or I close the brush window, it draws well, but after I open a brush group, I see a significant slowdown. 

Version: 1.2.0-beta.3+gitexport.0428646
Tablet used: Wacom Intuos CTH-480

**Comment by achadwick** [comment]
Interesting (see also #390 for other Windows lag issues, although the tread's getting a bit unmanagable now).

Which build is this - one from https://github.com/mypaint/mypaint/releases ?

1. Does this refer to brush group panels docked in the sidebar, or popped out into a floating window, or both?
2. Does painting speed improve when MyPaint is fullscreened (F11), after any sidebars or floating windows containing brush group panels auto-hide?
3. Does painting speed completely return to normal after the brush group panel is closed?
4. Do more brush group panels cause more slowdown?
5. Is this reproducible in Windows 7?
6. Are you using the "w32" or "w64" build, if it's one of ours?

**Comment by Bobhostern** [comment]
Build is 1.2.0-beta.3+gitexport.0428646
1. Both
2. Yes, it improves when the brush panel is auto-hid in fullscreen.
3. Yes.
4. Yes.
5. I'm not sure, I don't have Windows 7...
6. I'm using a w64 build.

**Comment by f-o-a-l-y** [comment]
I've got the same problem.
1.2.0-beta.3+gitexport.0428646

I'm using MyPaint w64 on Windows 7.
I don't remember any lag with version 1.2.0-beta.1.

I also experienced that sometimes the brush group panels will close with a long delay (> 10 s) after clicking the little "x" on them, sometimes they don't close at all.
And if they weren't docked but in their own windows, those windows will sometimes stay open after I closed the brush panel. So basically I then have a bunch of empty windows. These empty windows also re-open when restarting MyPaint. (I tried, but I somehow can't reproduce that right now.)

**Comment by ashleymarie93** [comment]
I am having the same lag issue with the brush group window open in 1.2.0-beta.3 version
I use windows 10

I also had an issue closing the brush group window. As well as when I tried to drag the window [to make it unconnected from the side] I got an extra window that said "Python" at the top instead of the brush group name and could not close that window either.

I then received an Error message:

```python
Mypaint version: 1.2.0-beta.3+gitexport.0428646
System information: Windows-10.0.10240
Traceback (most recent call last):
  File "C:\Program Files\MyPaint\share\mypaint/lib/document.py", line 418, _cache_updater_cb(self=<Document nlayers=2 bbox=Rect(-64, -128, 2432, 1280) paintonly=False>)
            activity_file_path = os.path.join(self.cache_dir, CACHE_ACTIVITY_FILE)
            os.utime(activity_file_path, None)
            if self._autosave_dirty:
  variables: {'activity_file_path': ('local', u'C:\\Users\\Ashley\\AppData\\Local\\Microsoft\\Windows\\INetCache\\mypaint\\doc.smynvk\\active'), 'None': ('builtin', None), 'os.utime': ('global', <built-in function utime>)}
WindowsError: [Error 2] The system cannot find the file specified: u'C:\\Users\\Ashley\\AppData\\Local\\Microsoft\\Windows\\INetCache\\mypaint\\doc.smynvk\\active'
```  

I wasn't sure if they were related but I closed MyPaint and re opened it and since have not had a problem closing the window or dragging it.
I don't code and it could just be my computer, I have had some issues with it lately. But I figured I  would give feedback since I really love this program and want to see it improve.

Thanks 
Ashley

**Comment by achadwick** [comment]
@ashleymarie93 (or anyone else) If you see the "WindowsError: [Error 2] The system cannot find the file specified" message mentioning `activity_file_path` again, please report it as a separate issue. Ideally with a way to reproduce it. Thanks! (:thought_balloon: It's probably unrelated to this issue. :fearful: Also why would bits of the profile area just disappear like that? :alien:)

**Comment by achadwick** [comment]
#514 may be related to this issue, though the symptoms are a little different. Though without affected hardware, I can't really test it.

If anyone affected by this bug is able to test by buildinf from source ([instructions here](https://github.com/mypaint/mypaint/blob/master/README_WINDOWS.md)), the most recent revisions on the development master branch contain a small fix for what I speculated about [here in #514](https://github.com/mypaint/mypaint/issues/514#issuecomment-160416753).

**Comment by achadwick** [comment]
@Bobhostern @ashleymarie93 @f-o-a-l-y 
[MyPaint v1.21](https://github.com/mypaint/mypaint/releases/tag/v1.2.1) is now available. Can you please download and install, and test to see whether this bug is still present in it? Thanks.

**Comment by zb13y** [comment]
Reproduced with v1.2.0-beta.4.
Opening 3 brush panels in row cause significant lags. 2 brush panels less lag.

Fixed in 1.2.1+gitexport.bcf5a28d.
Now i can open 5 and more panels without lags.

**Comment by achadwick** [comment]
Thanks, @zb13y !
